ziti/internal/edge_impl/conn.go

```diff
@@ -298,4 +382,14 @@ func (conn *edgeConn) newChildConnection(event *edge.MsgEvent) {
 	} else {
 		logger.Errorf("failed to receive start after dial. got %v", startMsg)
 	}
+
+	clientKey := message.Headers[edge.PublicKeyHeader]
+	if clientKey != nil {
+		err = edgeCh.establishCrypto(conn.keyPair, clientKey, true)
+		if err != nil {
+			logger.Errorf("failed to establish crypto session %v", err)
+		}
+	} else {
+		logger.Warnf("client did not send its key. connection is not end-to-end encrypted")
```

Comment:
What is the scenario where this would be ok?

Reply:
when the service is not hosted (egress from a router)